kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/SessionManager.java

```diff
@@ -185,21 +202,102 @@ public void createEnvironment(Session session) {
     public void start() {
         Logger log = session.getLogger();
         log.status("Using Kubernetes at: " + client.getMasterUrl());
-        createNamespace(session);
+        createNamespace();
+        setupConsoleListener();
 
         addShutdownHook();
         try {
-            createEnvironment(session);
+            createEnvironment();
         } catch (Throwable t){
           removeShutdownHook();
           throw t;
         }
     }
 
+    private void addConsole(final String podName) {
+        if (watchersMap.containsKey(podName))
+            return;
+
+        String className = session.getCurrentClassName();
+        String methodName = session.getCurrentMethodName();
+        String fileName = consoleLogPath;
+
+        if (Strings.isNullOrEmpty(className))
+            className = "NOCLASS";
+        fileName += String.format("/%s", className);
+
+        if (Strings.isNotNullOrEmpty(methodName))
+            fileName += String.format("-%s", methodName);
+        fileName += String.format("-%s.log", podName);
+
+        try {
+            final FileOutputStream stream = new FileOutputStream(fileName);
+            LogWatch lw = client.pods().inNamespace(session.getNamespace()).withName(podName).watchLog(stream);
+            watchersMap.put(podName, Arrays.asList(lw, stream));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(String.format("Error storing the console log for pod %s", podName), e);
+        }
+    }
+
+    private void delConsole(String podName) {
+        Collection<Closeable> lw = watchersMap.get(podName);
+        if (lw == null)
+            return;
+
+        watchersMap.remove(podName);
+        IoUtils.safeClose((Closeable[]) lw.toArray());
+    }
+
+    private void setupConsoleListener() {
+        if (!configuration.isEnvironmentConsoleLogCopyEnabled())
+            return;
+
+        consoleLogPath = configuration.getEnvironmentConsoleLogPath();
+        if (Strings.isNullOrEmpty(consoleLogPath))
+            consoleLogPath = String.format("%s/target/surefire-reports", System.getProperty("user.dir"));
```

Comment:
What if user is running this for `IntelliJ IDEA` & they didn't set property to change this default path?
 
I believe directory `target/surefire-reports`  will be missing in case of `IDEA` as it is not using surefire to run tests.

Reply:
If the dir does not exist then it will be created. See line 257.

new File(consoleLogPath).mkdirs();

Reply:
Yeah, this is only the default value. It can be overridden by the user. If you have a better default value, please tell me and I'll update the code.